### PR TITLE
feat/add new query getAllPaginatedOrdersVendor to orderService, updat…

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Add new query getAllPaginatedOrdersVendor to orderService
- Updated order statuses in constants, types
- Connected vendor orders page to backend with sorting, pagination and filter by order status
- Add error handling to vendor orders page
- Refactored error handling on admin users page
- Refactored role setting on admin users page and setting page number with role changing
- Fixed admin section title component backLinkHref
- Fixed on vendor orders page orders table component to use state in links to return back from order details page
- Refactored useSalesDataHook not to use orders with status rejected in calculations, updated vendor pie chart to show data for the whole period

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-101)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-08-09 в 01 27 51
Снимок экрана 2024-08-09 в 01 28 01
Снимок экрана 2024-08-09 в 01 28 10

### CODE REVIEW

**Review by Arseniia-Damaksina — APPROVED** [review]

File: src/pages/vendor/VendorOrdersPage/index.tsx [hunk 1]
```diff
@@ -55,23 +77,26 @@ function VendorOrdersPage() {
...
               <SortButton
                 title={t('usersAdmin.sortButton')}
                 onClick={handleClick}
               />
             </Box>
           </Grid>
           <Grid item xs={12} mt={4}>
-            {data && data.length > 0 ? (
+            {orders && orders.length > 0 ? (
               <OrdersTable
```
Comment (djanbolotov) [inline src/pages/vendor/VendorOrdersPage/index.tsx:89]: we can simplify "orders && orders.length > 0 = orders?.length"

File: src/common/constants.ts [hunk 2]
```diff
@@ -318,13 +319,13 @@ export const monthAndDayFormat = 'MMMM d';
...
 export const orderStatus = {
-  NEW: 'New order',
-  REJECTED: 'Rejected',
-  SENT: 'Sent',
-  RECEIVED: 'Received',
-  SENT_BACK: 'Sent back',
-  OVERDUE: 'Overdue',
-  RETURNED: 'Returned',
+  NEW: OrderStatus.NEW,
+  REJECTED: OrderStatus.REJECTED,
```
Comment (djanbolotov) [inline src/common/constants.ts]: why you redefine the same constants? the only difference is the name of this constants OrderStatus and orderStatus and one of them is enum. may be we can delete one of them?

File: src/pages/vendor/VendorOrdersPage/index.tsx (hunk 1 above)
Comment (MargaritaSukhova) [reply src/pages/vendor/VendorOrdersPage/index.tsx:89]: fixed

File: src/common/constants.ts (hunk 2 above)
Comment (MargaritaSukhova) [reply src/common/constants.ts]: deleted from constants, but it was used all over the project, had to change it to enum

**Review by djanbolotov — APPROVED** [review]

**Pull request merged**